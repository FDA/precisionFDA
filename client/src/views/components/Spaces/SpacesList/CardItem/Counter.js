import React from 'react'
import PropTypes from 'prop-types'


import { getSpacesIcon } from '../../../../../helpers/spaces'
import Icon from '../../../Icon'


const Counter = ({ type, counter }) => {
  const icon = getSpacesIcon(type)

  return (
    <div className="spaces-list-card-data__counter">
      <div>
        <Icon icon={icon} />
      </div>
      <div>{counter}</div>
    </div>
  )
}

Counter.propTypes = {
  type: PropTypes.string.isRequired,
  counter: PropTypes.number.isRequired,
}

export default Counter
