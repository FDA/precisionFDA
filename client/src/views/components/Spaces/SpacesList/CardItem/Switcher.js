import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames/bind'

import SpaceShape from '../../../../shapes/SpaceShape'
import ToggleSwitch from '../ToggleSwitch'


const Switcher = ({ space, lockToggleHandler }) => {
  const classes = classNames({
    'spaces-list-card-switcher': true,
    'spaces-list-card-switcher--active': !space.isLocked,
    'spaces-list-card-switcher--locked': space.isLocked,
  })

  const statusClasses = classNames(
    'spaces-list-card-switcher__status',
    `spaces-list-card-switcher__status--${space.status}`,
    `remediation-card-switcher-${space.status}`,
  )

  if (!space.hasLockLink) {
    return (
      <div className={classes}>
        <div className={statusClasses}>{space.status}</div>
      </div>
    )
  }

  return (
    <div className={classes}>
      <div className="spaces-list-card-switcher__status spaces-list-card-switcher__status--active remediation-card-switcher-active">active</div>
      <div className="spaces-list-card-switcher__toggle">
        <ToggleSwitch space={space} toggleHandler={lockToggleHandler} />
      </div>
      <div className="spaces-list-card-switcher__status spaces-list-card-switcher__status--locked remediation-card-switcher-locked">locked</div>
    </div>
  )
}

export default Switcher

Switcher.propTypes = {
  space: PropTypes.exact(SpaceShape),
  lockToggleHandler: PropTypes.func,
}
