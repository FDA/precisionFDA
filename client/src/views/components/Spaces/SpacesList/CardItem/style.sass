@import "../../../../../styles/variables.sass"

$title-text-fs: 20px
$bigger-text-fs: 18px

.spaces-list-card
  width: 700px
  min-height: 400px
  border: 2px solid $color-primary
  padding: 20px
  margin-bottom: 50px
  margin-right: 50px
  display: flex
  flex-direction: column
  justify-content: space-between

  &__header
    display: flex
    justify-content: space-between

  &__title
    font-size: 36px

    a
      color: $color-primary
      overflow: hidden
      text-overflow: ellipsis
      max-width: 420px
      display: block

      &:hover
        color: $color-link-hover

  &__desc
    display: -webkit-box
    justify-content: space-between
    margin-top: 10px
    font-size: $bigger-text-fs

    .pfda-tags-list
      margin-left: 20px

    &-text
      display: -webkit-box
      max-lines: 2
      text-overflow: ellipsis
      overflow: hidden
      -webkit-line-clamp: 2
      -webkit-box-orient: vertical

  &__date
    display: flex
    justify-content: space-between
    &-label
      font-weight: bold
    &-value
      margin-left: 10px

  &__body
    display: flex
    justify-content: space-between
    margin-top: 10px

  &-data
    width: 100%
    color: #fff
    background-color: #333
    padding: 15px

    &-margin
      margin-right: 20px

    a
      color: #fff

    &--shared
      margin-right: 0
      background-color: $color-public
      color: #333
      a
        color: #333

    &__title
      font-size: $title-text-fs
      font-weight: bold

      a
        text-decoration: underline

    &__status
      text-transform: uppercase
      padding: 2px 5px
      color: #333

      & .fa
        margin-right: 5px

    &__leaders
      min-height: 80px
      padding-bottom: 20px
      font-size: $bigger-text-fs

    &__counters
      display: flex
      justify-content: space-between
      text-align: center

      & .fa
        font-size: $title-text-fs

.spaces-list-card-switcher
  text-transform: uppercase
  display: flex
  line-height: 34px
  margin-right: 10px

  &__status
    color: #333
    &--active
      color: $color-success
    &--locked
      color: $color-danger
    &--unactivated
      color: $color-warning
    &--deleted
      color: $color-danger

  &--locked
    .spaces-list-card-switcher__status--active
      color: #333
  &--active
    .spaces-list-card-switcher__status--locked
      color: #333

  &__toggle
    margin: 0px 10px

.remediation-card-switcher
  &-unactivated
    color: #686664
  &-active
    color: #0f854c
  &-deleted
    color: #9c5e07
  &-locked
    color: #c7463d
