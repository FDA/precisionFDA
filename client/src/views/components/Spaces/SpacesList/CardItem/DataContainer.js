import React from 'react'
import PropTypes from 'prop-types'
import { Link } from 'react-router-dom'
import classNames from 'classnames/bind'

import Counter from './Counter'
import UserShape from '../../../../shapes/UserShape'
import SpaceShape from '../../../../shapes/SpaceShape'
import { SPACE_REVIEW } from '../../../../../constants'


const UserLink = ({ user, helperText }) => (
  <div className="spaces-list-card-data__lead">
    <a href={user.url}>
      {`${user.name} ${helperText}`}
    </a>
  </div>
)

const DataContainer = ({ space }) => {
  const typeTitle = (space.isPrivate) ? 'Private Area' : 'Shared Area'
  const classes = classNames({
    'spaces-list-card-data': true,
    'spaces-list-card-data--private': space.isPrivate,
    'spaces-list-card-data--shared': !space.isPrivate,
  })

  return (
    <div className={classes}>
      <div className="pfda-padded-b10">
        <div className="spaces-list-card-data__title">
          { space.links.show ?
            <Link to={`/spaces/${space.id}`}>{typeTitle}</Link> :
            <span>{typeTitle}</span>
          }
        </div>
      </div>

      <div className="spaces-list-card-data__leaders">
        { (space.isPrivate && space.hostLead) &&
          <UserLink user={space.hostLead} helperText="(Lead)" />
        }
        { (!space.isPrivate && space.hostLead) &&
          <UserLink user={space.hostLead} helperText={`(${space.type === SPACE_REVIEW ? 'Reviewer' : 'Host'} Lead)`} />
        }
        { (!space.isPrivate && space.guestLead) &&
          <UserLink user={space.guestLead} helperText={`(${space.type === SPACE_REVIEW ? 'Sponsor' : 'Space'} Lead)`} />
        }
      </div>

      <div className="spaces-list-card-data__counters">
        <Counter type="files" counter={space.counters.files} />
        <Counter type="apps" counter={space.counters.apps} />
        <Counter type="workflows" counter={space.counters.workflows} />
        <Counter type="jobs" counter={space.counters.jobs} />
        <Counter type="members" counter={space.counters.members} />
      </div>

    </div>
  )
}

export default DataContainer


DataContainer.propTypes = {
  space: PropTypes.exact(SpaceShape),
}

UserLink.propTypes = {
  user: PropTypes.shape(UserShape),
  helperText: PropTypes.string,
}
