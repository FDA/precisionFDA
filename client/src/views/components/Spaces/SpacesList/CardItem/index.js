import React from 'react'
import { Link } from 'react-router-dom'
import PropTypes from 'prop-types'

import SpaceListShape from '../../../../shapes/SpaceListShape'
import TagsList from '../../../TagsList'
import Switcher from './Switcher'
import DataContainer from './DataContainer'
import './style.sass'


const CardItem = ({ space, lockToggleHandler }) => (
  <div className="spaces-list-card">
    <div className="spaces-list-card__header">
      <div className="spaces-list-card__title">
        { space.shared.links.show ?
          <Link to={`/spaces/${space.shared.id}`}>{space.shared.name}</Link> :
          <span>{space.shared.name}</span>
        }
      </div>
      <div>
        <Switcher space={space.shared} lockToggleHandler={lockToggleHandler} />
      </div>
    </div>

    <div className="spaces-list-card__desc">
      <div className="spaces-list-card__desc-text">{space.shared.desc}</div>
      <div>
        <TagsList tags={space.shared.tags} />
      </div>
    </div>

    <div className="row pfda-mr-t10">
      <div className="pull-right">
        <div className="spaces-list-card__date">
          <div className="spaces-list-card__date-label">Created on:</div>
          <div className="spaces-list-card__date-value">{space.shared.createdAt}</div>
        </div>
        <div className="spaces-list-card__date">
          <div className="spaces-list-card__date-label">Modified on:</div>
          <div className="spaces-list-card__date-value">{space.shared.updatedAt}</div>
        </div>
      </div>
    </div>

    <div className="spaces-list-card__body">
      {(space.hasPrivate) && <DataContainer space={space.private} />}
      <DataContainer space={space.shared} />
    </div>
  </div>
)

export default CardItem

CardItem.propTypes = {
  space: PropTypes.exact(SpaceListShape),
  lockToggleHandler: PropTypes.func,
}
