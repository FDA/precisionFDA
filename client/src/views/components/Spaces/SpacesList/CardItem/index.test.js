import React from 'react'
import { shallow } from 'enzyme'

import CardItem from './'


describe('<CardItem />', () => {
  let space = null

  beforeEach(() => {
    space = {
      id: 100,
      hasPrivate: false,
      shared: {
        name: 'some name',
        isLocked: true,
        links: {
          show: 'some link',
          unlock: 'some link',
        },
      },
    }
  })

  it('matches snapshot', () => {
    const wrapper = shallow(<CardItem space={space} />)
    expect(wrapper).toMatchSnapshot()
  })
})
