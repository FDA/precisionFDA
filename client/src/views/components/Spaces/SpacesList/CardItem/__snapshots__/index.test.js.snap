// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardItem /> matches snapshot 1`] = `
<div
  className="spaces-list-card"
>
  <div
    className="spaces-list-card__header"
  >
    <div
      className="spaces-list-card__title"
    >
      <Link
        to="/spaces/undefined"
      >
        some name
      </Link>
    </div>
    <div>
      <Switcher
        space={
          Object {
            "isLocked": true,
            "links": Object {
              "show": "some link",
              "unlock": "some link",
            },
            "name": "some name",
          }
        }
      />
    </div>
  </div>
  <div
    className="spaces-list-card__desc"
  >
    <div
      className="spaces-list-card__desc-text"
    />
    <div>
      <TagsList
        tags={Array []}
      />
    </div>
  </div>
  <div
    className="row pfda-mr-t10"
  >
    <div
      className="pull-right"
    >
      <div
        className="spaces-list-card__date"
      >
        <div
          className="spaces-list-card__date-label"
        >
          Created on:
        </div>
        <div
          className="spaces-list-card__date-value"
        />
      </div>
      <div
        className="spaces-list-card__date"
      >
        <div
          className="spaces-list-card__date-label"
        >
          Modified on:
        </div>
        <div
          className="spaces-list-card__date-value"
        />
      </div>
    </div>
  </div>
  <div
    className="spaces-list-card__body"
  >
    <DataContainer
      space={
        Object {
          "isLocked": true,
          "links": Object {
            "show": "some link",
            "unlock": "some link",
          },
          "name": "some name",
        }
      }
    />
  </div>
</div>
`;
